session/session.go

```diff
@@ -938,9 +938,20 @@ func (s *session) DropPreparedStmt(stmtID uint32) error {
 	return nil
 }
 
-func (s *session) Txn() kv.Transaction {
-	if !s.txn.Valid() {
-		return nil
+func (s *session) Txn(opt ...struct{}) kv.Transaction {
```

Comment:
How about changing the function signature to:
```go
func (s *session) Txn(lazy ...bool) kv.Transaction {
}
```

In https://github.com/pingcap/tidb/pull/8260/files#diff-350127760839dbfd52d23927f7ff2d95R245, we can change the function call to:
```go
if sctx.Txn(true).Valid() {
}
```

In next PR, change the function signature to:
```go
func (s *session) Txn(lazy bool) kv.Transaction {
}
```

And modify other `Txn()` function calls to `Txn(false)`

Reply:
I agree, opt struct{} is weird to me.

Reply:
`func (s *session) Txn(opt ...struct{}) kv.Transaction` is a temporary solution, I know it's ugly.

However, for the `Txn(lazy ...bool)` case, it's not better:

```
func Txn(lazy ...bool) {
    if len(lazy) != nil {
       // should we check lazy[0] == true or lazy[0] == false ?
    }
}
```
@zz-jason @winkyao

Reply:
```go
if len(lazy) > 1 {
	panic("go check your code");
}
if len(lazy) == 0 || !lazy[0] {
	// do the default thing
}
// do the lazy thing
```

Reply:
I've changed the function signature